hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/JpaMetamodelImpl.java

```diff
@@ -515,7 +515,7 @@ public void processJpa(
 			PersistentClass persistentClass,
 			MetadataContext context,
 			TypeConfiguration typeConfiguration) {
-		final Class javaType = persistentClass.getMappedClass();
+		final Class javaType = persistentClass.getMappedClass() == null ? Map.class : persistentClass.getMappedClass();
```

Comment:
It's been a while but I seem to recall that `PersistentClass#getMappedClass` would return `Map.class` rather than `null` if the implementation was map-mode.  There is likely a better way to resolve this, so I leave that up to you guys on what's the best approach here.